server/rest.js

```diff
@@ -0,0 +1,321 @@
+
+            const jsonResponse = {
+                id: `cmpl`,
+                object: "text_completion",
+                created: Date.now(),
+                model: req.body.model,
+                choices: [
+                {
+                    text: resultText,
+                    index: 0,
+                    logprobs: null,
+                    finish_reason: "stop"
+                }
+                ],
+            };
+
+            if (Boolean(req.body.stream)) {
+                jsonResponse.id = `cmpl-${resultText}`;
+                jsonResponse.choices[0].finish_reason = null;
+                //jsonResponse.object = "text_completion.chunk";
+
+                const subscription = processIncomingStream(resultText, res, jsonResponse);
+            } else {
+                const requestId = uuidv4();
+                jsonResponse.id = `cmpl-${requestId}`;
+                res.json(jsonResponse);
+            };
+        });
+        
+        app.post('/v1/chat/completions', async (req, res) => {
+            const modelName = req.body.model || 'gpt-3.5-turbo';
+            const pathwayName = openAIChatModels[modelName] || openAIChatModels['*'];
+
+            if (!pathwayName) {
+                res.status(404).json({
+                    error: `Model ${modelName} not found.`,
+                });
+                return;
+            }
+
+            const pathway = pathways[pathwayName];
+
+            const resultText = await processRestRequest(server, req, pathway, pathwayName);
+
+            const jsonResponse = {
+                id: `chatcmpl`,
+                object: "chat.completion",
+                created: Date.now(),
+                model: req.body.model,
+                choices: [
+                    {
+                        message: {
+                            role: "assistant",
+                            content: resultText
+                        },
+                        index: 0,
+                        finish_reason: "stop"
+                    }
+                ],
+            };
+
+            if (Boolean(req.body.stream)) {
+                jsonResponse.id = `chatcmpl-${resultText}`;
+                jsonResponse.choices[0] = {
+                    delta: {
+                        role: "assistant",
+                        content: resultText
+                    },
+                    finish_reason: null
+                }
+                jsonResponse.object = "chat.completion.chunk";
+
+                const subscription = processIncomingStream(resultText, res, jsonResponse);
+            } else {
+                const requestId = uuidv4();
+                jsonResponse.id = `chatcmpl-${requestId}`;
+
+                res.json(jsonResponse);
+            }
+
+        });
+
+        app.get('/v1/models', async (req, res) => {
+            const openAIModels = { ...openAIChatModels, ...openAICompletionModels };
+            const defaultModelId = 'gpt-3.5-turbo';
```

Comment:
should we do sth about this default guy

Reply:
We have to have at least one model available.  I suppose we could make it a config parameter, but this seems clear enough.  You only see this code when there are no other specific handlers (e.g. only wildcard handlers).